HealthMonitoring.SelfHost/Content/Scripts/homeController.js

```diff
@@ -122,26 +196,135 @@
                 Version: "-- connection issue --"
             };
         };
-
+        
         var getEndpoints = function () {
             healthmonitorEndpointService.getEndpoints().then(onEndpointsLoaded, onEndpointsFailed);
-        }
+        };
 
         var getConfig = function () {
             healthmonitorEndpointService.getConfig().then(onConfigLoaded, onConfigFailed);
-        }
+        };
 
         getEndpoints();
         getConfig();
 
         setInterval(getEndpoints, endpointFrequency);
         setInterval(getConfig, configFrequency);
-    }
+    };
 
     app.controller("HomeController", ["$scope", "$location", "healthmonitorEndpointService", homeController]);
 
 }());
 
+function getGroupsInfo(data) {
+
+    var groupsInfo = {};
+
+    angular.forEach(data, getGroupInfo);
+
+    var groups = [];
+
+    for (var key in groupsInfo) {
+        if (groupsInfo.hasOwnProperty(key)) {
+            groups.push(groupsInfo[key]);
+        }
+    }
+
+    return groups;
+
+    function getGroupInfo(item) {
+
+        if (groupsInfo.hasOwnProperty(item.Group)) {
```

Comment:
Actually this block can be reworked in two ways
1. (Simplest one) Add class Group with fields and initialize it with defaults via constructor, so we remove code from else block
2. Rethink logic of showing info from groups so we can simple remove it and aggregate data from groups